ra/registration-authority.go

```diff
@@ -176,29 +182,30 @@ func (ra *RegistrationAuthorityImpl) NewCertificate(req core.CertificateRequest,
 	if len(names) == 0 {
 		err = core.UnauthorizedError("CSR has no names in it")
```

Comment:
I feel it might make sense if this was a `MalformedRequestError`, although there is some cross over.